packages/backend/src/tokens/request.ts

```diff
@@ -188,6 +203,59 @@ ${error.getFullMessage()}`,
     return signedOut(authenticateContext, reason, message);
   }
 
+  // handleMaybeOrganizationSyncHandshake determines if a handshake must occur to resolve a mismatch between
+  // the organization as specified by the URL (according to the options) and the actual active organization
+  // on the session.
+  // NOTE(izaak): I don't love that null return implies signed-in...
+  function handleMaybeOrganizationSyncHandshake(
+    authenticateContext: AuthenticateContext,
+    auth: SignedInAuthObject,
+  ): HandshakeState | SignedOutState | null {
+    if (!options.organizationSync) {
+      return null;
+    }
+    const toActivate = getActivationEntity(authenticateContext.clerkUrl, options.organizationSync);
+    if (!toActivate) {
+      return null;
+    }
+    let mustActivate = false;
+    // Activate an org by slug?
+    if (toActivate.organizationSlug && toActivate.organizationSlug !== auth.orgSlug) {
+      mustActivate = true;
+    }
+    // Activate an org by ID?
+    if (toActivate.organizationId && toActivate.organizationId !== auth.orgId) {
+      mustActivate = true;
+    }
+    // Activate the personal workspace?
+    if (toActivate.type === 'personalWorkspace' && auth.orgId) {
+      mustActivate = true;
+    }
+    if (!mustActivate) {
+      return null;
+    }
+    if (authenticateContext.handshakeRedirectLoopCounter > 0) {
+      // We have an organization that needs to be activated, but this isn't our first time redirecting.
+      // This is because we attempted to activate the organization previously, but the organization
+      // must not have been valid (either not found, or not valid for this user), and gave us back
+      // a null organization. We won't re-try the handshake, and leave it to the server component to handle.
+      console.warn(
+        'Clerk: Organization activation handshake loop detected. This is likely due to an invalid organization ID or slug. Skipping organization activation.',
```

Comment:
Let's move this to `errors.ts`.

Reply:
We chatted about this, and decided to leave it. Rationalle:
- Other logs in this file (and in the backend package generally) aren't declaring error consts
- This error is very specific and unlikely to be reused